packages/suite/src/components/suite/modals/ReduxModal/UserContextModal/TxDetailModal/AdvancedTxDetails/IODetails/IODetails.tsx

```diff
@@ -211,18 +214,37 @@ const EthereumSpecificBalanceDetailsRow = ({
                         />
                     </H4>
                     {tokens.map((transfer, index) => {
-                        const value = isNftTokenTransfer(transfer) ? (
-                            <FormattedNftAmount transfer={transfer} isWithLink />
-                        ) : (
-                            formatAmount(transfer.amount, transfer.decimals)
+                        const value = (
+                            <AmountComponent
+                                transfer={transfer}
+                                useStyled={false}
+                                withLink={true}
+                                contract={transfer.contract as TokenAddress}
+                                withSign={false}
+                                alignMultitoken="flex-start"
+                            />
                         );
 
                         return (
                             <IOGroup
                                 key={index}
                                 tx={{ ...tx, symbol: transfer.symbol as NetworkSymbol }}
-                                inputs={[{ addresses: [transfer.from], value }] as IODetails[]}
-                                outputs={[{ addresses: [transfer.to], value }] as IODetails[]}
+                                inputs={
+                                    [
+                                        {
+                                            addresses: [transfer.from],
+                                            value: value as ReactNode,
+                                        },
+                                    ] as IODetails[]
+                                }
+                                outputs={
+                                    [
+                                        {
+                                            addresses: [transfer.to],
+                                            value: value as ReactNode,
+                                        },
+                                    ] as IODetails[]
+                                }
```

Comment:
nitpick: I would suggest extract these arrays to consts above. It's easier to read

Reply:
I actually returned it to the way it was before